[Test QoS] macOS needs target_link_libraries GTEST_LIBRARIES

## Bug report

Compiliing [ROS2 Iron from source on macOS](https://docs.ros.org/en/iron/Installation/Alternatives/macOS-Development-Setup.html) gets an error after fded516933cca6f5bafc64bac489c86dfee4e323.

**Required Info:**

- Operating System:
  - macOS 13.4 Ventura (arm64)
- Installation type:
  - from source
- Version or commit hash:
  - iron @ 6324429e11b13287e77269c6ab39d17509ec77ce
- DDS implementation:
  - default: Fast DDS
- Client library (if applicable):
  - N/A

#### Steps to reproduce issue

```console
# follow ROS2 iron compile from source instructions

mkdir -p ~/ros2_iron/src
cd ~/ros2_iron
vcs import --input https://raw.githubusercontent.com/ros2/ros2/iron/ros2.repos src

cd ~/ros2_iron/
colcon build --symlink-install --packages-skip-by-dep python_qt_binding
```

#### Expected behavior

* system_tests compiles like other repositories.

#### Actual behavior

```console
--- stderr: test_quality_of_service                                                                                                                                                                                          
Undefined symbols for architecture arm64:
  "typeinfo for testing::Test", referenced from:
      typeinfo for BaseQosRclcppTestFixture in qos_utilities.cpp.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [libtest_quality_of_service_support.dylib] Error 1
make[1]: *** [CMakeFiles/test_quality_of_service_support.dir/all] Error 2
make: *** [all] Error 2
make: INTERNAL: Exiting with 11 jobserver tokens available; should be 10!
---
Failed   <<< test_quality_of_service [11.5s, exited with code 2]
```

#### Additional information

Re-inserting `target_link_libraries(${PROJECT_NAME}_support ${GTEST_LIBRARIES})` (test_quality_of_service/CMakeLists.txt:47), makes it compile. Maybe we should re-add it or re-add it using `if (APPLE)`.

--- a/test_quality_of_service/CMakeLists.txt
+++ b/test_quality_of_service/CMakeLists.txt
@@ -39,11 +39,9 @@ if(BUILD_TESTING)
     test/qos_test_node.cpp
     test/qos_test_publisher.cpp
     test/qos_test_subscriber.cpp
-    test/qos_utilities.cpp
   )
 
   ament_find_gtest()
-  target_include_directories(${PROJECT_NAME}_support PUBLIC "${GTEST_INCLUDE_DIRS}")
   target_compile_definitions(${PROJECT_NAME}_support PRIVATE
     "TEST_QUALITY_OF_SERVICE_BUILDING_LIBRARY")
   ament_target_dependencies(${PROJECT_NAME}_support rclcpp rcutils std_msgs)
@@ -61,10 +59,10 @@ if(BUILD_TESTING)
   endfunction()
 
   macro(tests)
-    add_custom_gtest(test_deadline test/test_deadline.cpp)
-    add_custom_gtest(test_lifespan test/test_lifespan.cpp)
-    add_custom_gtest(test_liveliness test/test_liveliness.cpp)
-    add_custom_gtest(test_best_available test/test_best_available.cpp)
+    add_custom_gtest(test_deadline test/test_deadline.cpp test/qos_utilities.cpp)
+    add_custom_gtest(test_lifespan test/test_lifespan.cpp test/qos_utilities.cpp)
+    add_custom_gtest(test_liveliness test/test_liveliness.cpp test/qos_utilities.cpp)
+    add_custom_gtest(test_best_available test/test_best_available.cpp test/qos_utilities.cpp)
   endmacro()
 
   call_for_each_rmw_implementation(tests)

--- a/test_quality_of_service/include/test_quality_of_service/qos_utilities.hpp
+++ b/test_quality_of_service/include/test_quality_of_service/qos_utilities.hpp
@@ -40,7 +40,6 @@
  * @param milliseconds
  * @return tuple of milliseconds converted to <seconds, nanoseconds>
  */
-TEST_QUALITY_OF_SERVICE_PUBLIC
 std::tuple<size_t, size_t> convert_chrono_milliseconds_to_size_t(
   const std::chrono::milliseconds & milliseconds);
 
@@ -67,9 +66,7 @@ bool wait_for(
 class BaseQosRclcppTestFixture : public ::testing::Test
 {
 protected:
-  TEST_QUALITY_OF_SERVICE_PUBLIC
   void SetUp() override;
-  TEST_QUALITY_OF_SERVICE_PUBLIC
   void TearDown() override;
   // executor used to submit work
   std::shared_ptr<rclcpp::executors::SingleThreadedExecutor> executor;

